src/datumaro/plugins/anchor_generator.py

```diff
@@ -0,0 +1,388 @@
+                else:
+                    init_scales.append(
+                        [
+                            torch.kthvalue(
+                                torch.Tensor(scale_samples), int(p * len(scale_samples))
+                            ).values.item()
+                            for p in representatives
+                        ]
+                    )
+                    init_ratios.append(
+                        [
+                            torch.kthvalue(
+                                torch.Tensor(ratio_samples), int(p * len(ratio_samples))
+                            ).values.item()
+                            for p in representatives
+                        ]
+                    )
+
+            return torch.Tensor(init_scales), torch.Tensor(init_ratios)
+
+        def get_loss(self, targets: torch.Tensor, scales: torch.Tensor, ratios: torch.Tensor):
+            """This function is to create base anchors, which combinates ratios and scales.
+
+            Args:
+                targets (Tensor): Set of target bounding box coordinates.
+                scales (Tensor): Scales for creating base anchors.
+                ratios (Tensor): Ratios for creating base anchors.
+
+            Returns:
+                float: Cost.
+                float: Coverage rate.
+            """
+            all_anchors = []
+            for level, base_size in enumerate(self.strides):
+                anchors = self.get_anchors(
+                    base_size, self.shifts[level], scales[level], ratios[level]
+                )
+                all_anchors.append(anchors)
+            all_anchors = torch.cat(all_anchors, dim=0)
+
+            overlaps = self.iou_calculator(targets, all_anchors)
+
+            max_overlaps, _ = overlaps.max(dim=0)
+            log.info(
+                f"[ANCHOR] total: {overlaps.shape[1]}, "
+                f"pos: {sum(max_overlaps >= self.pos_thr)}, "
+                f"neg: {sum(max_overlaps < self.neg_thr)}"
+            )
+
+            gt_max_overlaps, _ = overlaps.max(dim=1)
+            log.info(
+                f"[GT] total: {overlaps.shape[0]}, "
+                f"pos: {sum(gt_max_overlaps >= self.pos_thr)}, "
+                f"neg: {sum(gt_max_overlaps < self.neg_thr)}"
+            )
+
+            cost = (
+                500 * gt_max_overlaps[(gt_max_overlaps >= self.pos_thr)].sum()
+                + max_overlaps[(max_overlaps >= self.pos_thr)].sum()
+                - max_overlaps[(max_overlaps < self.neg_thr)].sum()
+            )
+            log.info(f"Cost: {cost}")
+
+            num_gts = overlaps.size(0)
+            pos_gts = 0
+            for i in range(num_gts):
+                if gt_max_overlaps[i] >= self.pos_thr:
+                    max_iou_inds = (overlaps[i, :] == gt_max_overlaps[i]).nonzero()
+                    if max_iou_inds.numel() != 0:
+                        overlaps[:, max_iou_inds[0]] = 0
+                        pos_gts += 1
```

Comment:
```suggestion
            pos_gts = (gt_max_overlaps >= self.pos_thr).sum()
```
I'm not sure if this could be converted according to these suggestions, but it seems similar to them. If not, please use the original version

Reply:
Unfortunately, they are not same. I want to have unique pairs between target and anchor at there.